pkg/kubelet/server/server.go

```diff
@@ -231,7 +233,7 @@ func NewServer(
 	}
 	server.InstallDefaultHandlers(enableCAdvisorJSONEndpoints)
 	if enableDebuggingHandlers {
-		server.InstallDebuggingHandlers(criHandler)
+		server.InstallDebuggingHandlers(criHandler, enableSystemLogHandler)
```

Comment:
It seems inappropriate for enableDebuggingHandlers to override enableSystemLogHandler.  We can't change the old behavior of the flags, but having the handler be split into its own specific behavior would be better.

If a user says `enableSystemLogHandler=true` that should probably override `enableDebuggingHandlers=false`

Reply:
Sure, sounds good to me. I will update the PR accordingly.

Reply:
Sure sounds good to me. will update this PR.

Reply:
> It seems inappropriate for enableDebuggingHandlers to override enableSystemLogHandler. We can't change the old behavior of the flags, but having the handler be split into its own specific behavior would be better.
> 
> If a user says `enableSystemLogHandler=true` that should probably override `enableDebuggingHandlers=false`

@liggitt this was the initial suggestion/discussion when we decided to separate systemLogHandler from DebuggingHandler.

Reply:
> If a user says enableSystemLogHandler=true that should probably override enableDebuggingHandlers=false

The issue is defaulting and compatibility... an existing setup that disables logs by setting enableDebuggingHandlers=false should still work.

Then, there are three paths to setting kubelet config values:
1. CLI flags
2. config file fields
3. dynamic config file fields

Normal API defaulting breaks down because of the way these three can be combined. CLI flags override config file fields, so cross-field defaulting (e.g. default `enableSystemLogHandler` to the value of `enableDebuggingHandlers`) doesn't work properly.

Reply:
I agree that this makes both logic and code simpler but worth noting that in future when we separate other endpoints from debugging handler like pprof (https://github.com/kubernetes/kubernetes/pull/84397) or other endpoints then we have to follow this path where debuggingHandler has to be enabled.